src/inference/particle_filter.jl

```diff
@@ -111,7 +111,29 @@ end
     particle_filter_step!(state::ParticleFilterState, new_args::Tuple, argdiffs,
         observations::ChoiceMap, proposal::GenerativeFunction, proposal_args::Tuple)
 
-Perform a particle filter update, where the model arguments are adjusted, new observations are added, and some combination of a custom proposal and the model's internal proposal is used for proposing new latent state (whatever is not proposed from the custom proposal will be proposed using the model's internal proposal).
+Perform a particle filter update, where the model arguments are adjusted, new
+observations are added, and some combination of a custom proposal and the
+model's internal proposal is used for proposing new latent state.  That is, for
+each particle,
+
+* The proposal function `proposal` is evaluated with arguments `Tuple(t_old,
+  proposal_args...)`, and produces a trace (call it `proposal_trace`); and
+* The old trace (call it `t_old`) is replaced by a new trace (call it `t_new`).
+
+The choicemap of `t_new` satisfies the following conditions:
+
+1. `get_choices(t_old)` is a subset of `get_choices(t_new)`;
+2. `observations` is a subset of `get_choices(t_new)`;
+3. `get_choices(proposal_trace)` is a subset of `get_choices(t_new)`.
+
+Here, when we say one choicemap `a` is a "subset" of another choicemap `b`, we
+mean that all keys that occur in `a` also occur in `b`, and the values at those
+addresses are equal.
+
+It is an error if no trace `t_new` satisfying the above conditions exists in
+the support of the model (with the new arguments). If such a trace exists, then
+the random choices not determined by the above requirements are sampled using
+the internal proposal.
```

Comment:
@marcoct, @alex-lew, updated the docstring based on your feedback -- please take a look when you get the chance

Reply:
Looks good to me!

Reply:
Me too!